rotkehlchen/db/eth2.py

```diff
@@ -169,7 +171,17 @@ def get_validator_daily_stats(
         query, bindings = filter_query.prepare()
         query = 'SELECT * from eth2_daily_staking_details ' + query
         results = cursor.execute(query, bindings)
-        return [ValidatorDailyStats.deserialize_from_db(x) for x in results]
+        daily_stats = [ValidatorDailyStats.deserialize_from_db(x) for x in results]
+        # Take into account the proportion of the validator owned
+        validators_ownership = {
+            validator.index: validator.ownership_proportion
+            for validator in self.get_validators()
+        }
+        for i, daily_stat in enumerate(daily_stats):
+            owned_proportion = validators_ownership.get(daily_stat.validator_index, FVal(1))
+            if owned_proportion != ONE:
+                daily_stats[i] = daily_stat._replace(pnl=daily_stat.pnl * owned_proportion)
```

Comment:
Argh this sucks. Well it's no longer an immutable named tuple with this functionality. So please change `ValidatorDailyStats` to a dataclass with frozen=False. Then you will no longer need to use the `_replace` hack